llvm/CMakeLists.txt

```diff
@@ -149,6 +149,12 @@ if ("flang" IN_LIST LLVM_ENABLE_PROJECTS)
     message(STATUS "Enabling clang as a dependency to flang")
     list(APPEND LLVM_ENABLE_PROJECTS "clang")
   endif()
+
+  option(FLANG_ENABLE_FLANG_RT "Implicitly add LLVM_ENABLE_RUNTIMES=flang-rt when compiling Flang" ON)
+  if (FLANG_ENABLE_FLANG_RT AND NOT "flang-rt" IN_LIST LLVM_ENABLE_RUNTIMES)
+    message(STATUS "Enabling Flang-RT as a dependency of Flang")
+    list(APPEND LLVM_ENABLE_RUNTIMES "flang-rt")
```

Comment:
I believe you can't `list(APPEND ..)` cache variables like this as they'll get overridden by what was provided on the command line. I think it should be `set(LLVM_ENABLE_RUNTIMES "${LLVM_ENABLE_RUNTIMES};flang-rt" CACHE INTERNAL "")` or something along those lines

Reply:
Hmm, that doesn't seem to work either. I actually don't see the message so for some reason we aren't entering that if statement body I guess?

Reply:
Sorry. All of that was false alarms. The issue is that LLVM_ENABLE_RUNTIMES hasn't been defined by this point, it's defined a little after these checks. When it does get defined it overwrites whatever was done here. 

If you move the LLVM_ENABLE_RUNTIMES define above this, then this works as expected.